build-docker.sh

```diff
@@ -15,12 +15,12 @@ if [[ "${CIRCLE_BRANCH}" == "main" ]]; then
     TAGS+=",latest"
 fi
 
-# Local builds just get "dev" for version
-VERSION="dev"
+# Determine a version number based on most recent version tag in git.
+VERSION=$(git describe --tags --match='v[0-9]*' --always)
 
-# If we are doing a dev build on circle, we will version it as the circleci buildnumber
-if [[ -n "${CIRCLE_BUILD_NUM}" ]]; then
-    VERSION="${CIRCLE_BUILD_NUM}"
+# If we are doing a dev build on circle, append the build number (job id) to the version
+if [[ -n "${CIRCLE_BUILD_NUM:-}" ]]; then
+  VERSION="${VERSION}-ci${CIRCLE_BUILD_NUM}"
```

Comment:
Personally, I don't see the value in the 45. I also like dropping g and ci, but I don't think we win much with the + and _ -- I'd rather see dashes or underscores everywhere.

Reply:
I like the idea to quickly know how much deviation between a dev build and the last stable release. Since the CI build ID is always a number, I find it helpful to have the `ci` prefix to remind me what the number is. However, I agree that the `g` for git is not as helpful since commit hash also contains letters.

Reply:
Starting with defaults, the `45` and the `g` is what comes out of the `git describe` command:

```
❯ git describe --tags --match='v[0-9]*' --always
v2.8.2-45-ge527ea1
```
> `g` for git is not as helpful since commit hash also contains letters

I agree that's annoying. I happen to know that commit IDs are SHA1 hashes and SHA1 hashes are hexadecimal and `g` is not a hex digit, so my brain parses that fine and it doesn't bother me _that_ much. … But as I list all that out, I see that it takes me knowing a lot of things to get to the point of it not bothering me much.

So my _current_ tolerance for this format is such that I am _currently_ not convinced that the added complexity to shape this dev version differently will be worth writing and maintaining. I am open to being convinced, though.

Reply:
I think it's not worth fixing that for a minor annoyance. I'm going to approve this, but it might be useful to also hear from others before merging.

Reply:
Oh, that's makes much more sense. Thank you for the explanation. My brain can now process it better. Always learning something new from Robb. 

I agree with you that this is not something that worth adding more complexity for

Reply:
Agreed, not worth the added effort. This gives me so much information so I'm plenty pleased 😃

Reply:
I also don't find much value in the ci prefix for circle build IDs. I'm don't mind the hyphen's or the number of commits as they match the default git format.

Also agree there's we've been blocked on this for awhile, we can always come back and change it if needed but this better than what we have now.